### ISSUE TITLE
rrules exceptions not managed

### ISSUE THREAD

**Issue by sherpadawan** [wernerjoss/grav-plugin-fullcalendar#18]
In case of a recurrent event, exceptions are notre managed.
I personnally did not succeed in displaying exceptions and not the event of the chain.

**Comment by wernerjoss** [comment]
you are right with that: EXDATE/EXRULE are not yet supported.
and, to be honest, this whole rrule thing never had much priority for me, as I don't need it for my personal use 😄 .
I just implemented basic rrule support upon user request so far.
so, l'll put this on the TODO list and have a closer look when I have time, but no guarantee when this will happen.
If you want to pick this up and come up with a solution in the meantime, I'll be glad to merge that.
cheers
Werner

**Comment by sherpadawan** [comment]
Ok but I think it is a tricky develpement since some calendar ics files do not have EXDATE value ...

**Comment by wernerjoss** [comment]
yes, that seems so - and I do not remember any ICS file with such an entry 😄 .
additionally, same holds for EXRULE, which is a whole rule set inside.
as said, I'll look into it when I have time.
in the meantime, I'm sure 99% of the grav fullcalendar users will not even notice this missing.

**Comment by sherpadawan** [comment]
I finally rewrite a big part of the plugin :
- ical expander integration of rrule integration
- jquery replacement by superagent node module
- more settings passed by the plugin to javascript (need to filter nore precisly for security reason)
- ics : page files, and http files supported (Shortcodes removed as it is not a good pratice)
- development : assets is a webpack project

Still need some work on the plugin. But take a look to my fork.

Regards.

**Comment by wernerjoss** [comment]
@sherpadawan : yes, your fork seems to be a big rewrite 😄 .
I just took a quick look and installed it on a docker instance to check it out.
installed grav events and taxonomy plugins, which seem to be necessary, created a calendar page and dropped an ics file in it.
calendar grid is then displayed, but in console I get this:
![grafik](https://user-images.githubusercontent.com/36265757/90128089-9128e300-dd66-11ea-92a1-699181e9862c.png)
now I'm not sure if this is due to missing/wrong configuration, I'll have a closer look next week, as I don't have more time ATM.
cheers
Werner

**Comment by sherpadawan** [comment]
Hello !

I pushed changes in the repostory like rewritten documentation, and some 
fixes.

1) Taxonomy

One need to set header page in expert mode

taxonomy:
     category:
         - calendar

or in the page form options tab in normal mode

2) templates

One can use fullcalendar template or modular one

3) events are cached in local storage but it does not seem to expire 
despite page reload setting ...

ttyl

Guillaume


Le 13/08/2020 à 13:13, Werner Joss a écrit :

**Comment by sherpadawan** [comment]
Hello !

I just released a new version. Do you want to  merge it ?

Thanks in advance

Le 25/08/2020 à 09:20, Guillaume Storchi a écrit :

**Comment by wernerjoss** [comment]
ok, I just pulled the new version, added the mentioned taxonometry to the calendar page (not a modular page),
dropped a calendar ics file in the content area, then I get this:
![grafik](https://user-images.githubusercontent.com/36265757/92019060-4e9b6a80-ed56-11ea-800e-278ac327cad3.png)
btw., I'm still on Grav 1.6.26.

**Comment by sherpadawan** [comment]
Ok thanks for your test. I give you some feedback soon

Le 02/09/2020 à 19:57, Werner Joss a écrit :

**Comment by sherpadawan** [comment]
Ok you can test it now. 0.3.1

taxonomy is not required anymore.

Sorry for the delay. I tested it on modular pages with modular fullcalendar submodule, and on a regular fullcalendar page.

Le 03/09/2020 à 09:05, Guillaume Storchi a écrit :

**Comment by wernerjoss** [comment]
Ok, now I can see a Calendar :smile: 
but only the last one, if I add more than 1 ics File - so currently, there can be only one calendar file per page.
old css does not work either, this is expected, as fullcalendar.io v5 has different classes than fc4.
show description/summary on hover is also not working, nor follow links when clicked on event that has an URL
(instead, there is a stale event entry just to the right).
but these are minor issues, I think.
I have now created an new branch, fc5, where I want to merge your version and see how I can get things like
month picture, legend... to work.
if you like, we can then collaborate on this branch until it is ready to replace master branch.
one big issue before we can do that is provide a (hopefully) automatic migration procedure for users of the current version.
just replacing the old plugin with fc5 branch will brick current users pages because of shortcuts no longer working, which is a no-go, IMHO.
what do you think ?

**Comment by sherpadawan** [comment]
Hi !

I am definitly not a friend of shortcodes. I won't work on it because of 
a lack of time.

In my opinion it is more interesting to develop the missing form in 
admin page tab (fullcalendar), and allow one to set parameters there 
than proposing a migration script for users. It is a very small work to 
migrate old shortcodes in pages to the new header way to configure a 
calendar.

It would help me if you post issues on my repo page directly, if it is 
ok for you.

I'll ake a look to multiple calendar issues.

Thx for your feedbacks.

Guillaume

Le 05/09/2020 à 17:05, Werner Joss a écrit :

**Comment by sherpadawan** [comment]
Ok for the fc5 branch.

Le 05/09/2020 à 17:05, Werner Joss a écrit :

**Comment by wernerjoss** [comment]
> Ok for the fc5 branch. 

ok, can you please create a new pull request for fc5 branch ?
I will then merge and then we can start from there.
I'll answer to your other reply tomorrow.
slt
Werner

**Comment by wernerjoss** [comment]
I think it will be better to just replace fc5 now with current state of your fork, and then add things from master as appropriate, as due to too many incompatible differences, a merge would not result in anything useful anyway.
what do you think about this ?

**Comment by sherpadawan** [comment]
I made a mistake, you can reject the README change on your repo... I thought I was on mine.

**Comment by wernerjoss** [comment]
hello again,
after awhile of no activitiy on the project, I have now released an new version: 0.2.6.
I like your idea of just drop ics file(s) - yes, more than one :smile: - in the calendar page, so it is now implemented.
I retained however the shortcode method, as dropping this would break the pages of existing users, which is a no-go.
another of your proposals: rip out the javascript code from the templates into extra javascript files, has also been implemented.
next step will be looking into replace fc4 with fc5, and eventually assets as a webpack project, using the 3rd party libraries you already proposed to fix the rrules exceptions.
